lib/AST/StmtPrinter.cpp

```diff
@@ -1735,6 +1735,46 @@ void StmtPrinter::VisitAtomicExpr(AtomicExpr *Node) {
   OS << ")";
 }
 
+// Checked C extension
+
+void StmtPrinter::VisitCountBoundsExpr(CountBoundsExpr *Node) {
+  switch (Node->getKind()) {
+    case CountBoundsExpr::Kind::Byte_Count:
+      OS << "byte_count(";
+    case CountBoundsExpr::Kind::Count:
+      OS << "count(";
+    default:
+      llvm_unreachable("count bounds kind not covered by switch");
+      break;
+  }
+  PrintExpr(Node->getCountExpr());
+  OS << ")";
+}
+
+void StmtPrinter::VisitNullaryBoundsExpr(NullaryBoundsExpr *Node) {
+  switch (Node->getKind()) {
+  case NullaryBoundsExpr::Invalid: {
+    OS << "bounds(invalid)";
+    break;
+  }
+  case NullaryBoundsExpr::None: {
+    OS << "bounds(none)";
+    break;
+  }
+  default:
+    llvm_unreachable("count bounds kind not covered by switch");
+    break;
+  }
+}
+
+void StmtPrinter::VisitRangeBoundsExpr(RangeBoundsExpr *Node) {
+  OS << "bounds(";
+  PrintExpr(Node->getLowerExpr());
+  OS << ",";
```

Comment:
The printer seems to use whitespace after commas -- we should follow that here and make this ", ".